### DISCUSSION TITLE
Difference between `UnitResult<T>` and `Result`

### DISCUSSION THREAD

**Discussion by lonix1** [vkhorikov/CSharpFunctionalExtensions#398] [Q&A] [answered]
A unit result has no value for the success case, but has an error for the failed case.

How does that differ to using a plain `Result`?

**Comment by vkhorikov** [comment, marked as answer]
`Result` is a shortcut for `Result<Unit, string>`, whereas `UnitResult` is a shortcut for `Result<Unit, T>`. The difference is that in `UnitResult` the error type may vary, whereas in `Result` it's set to `string`. See also this discussion: https://github.com/vkhorikov/CSharpFunctionalExtensions/issues/321

**Comment by xavierjohn** [comment]
Was there a real need to introduce `UnitResult` vs just a struct named `Unit` ?  
Function that has no result could just return Result<Unit, TE> and we could avoid a huge explosion of functions to support `UnitResult`

**Comment by vkhorikov** [comment]
I agree. I'm not sure if there was a 100% need for that, the topic itself is arguable. But in my projects, I just use `Result<Unit, T>`.
